CMakeLists.txt

```diff
@@ -140,41 +203,147 @@ SET ( NLOPT_SOURCES
-  STRING( REGEX REPLACE "SHARED_VERSION_INFO=\"[0-9]+:[0-9]+:([0-9]+)\"" "\\1" _VERSION_INFO_AGE "${_VERSION_INFO_LINE}") 
-  MATH(EXPR _INFO_SOVERSION "${_VERSION_INFO_CURRENT} - ${_VERSION_INFO_AGE} ")
-  SET(_INFO_VERSION "${_INFO_SOVERSION}.${_VERSION_INFO_AGE}.${_VERSION_INFO_REVISION}")
-  SET_TARGET_PROPERTIES(nlopt PROPERTIES SOVERSION ${_INFO_SOVERSION})
-  SET_TARGET_PROPERTIES(nlopt PROPERTIES VERSION ${_INFO_VERSION} )
-ENDIF ()
+# pass -fPIC in case swig module is built with static library
+if (NOT BUILD_SHARED_LIBS)
+  check_c_compiler_flag (-fPIC HAS_FPIC)
+  if (HAS_FPIC)
+    set (CMAKE_C_FLAGS "-fPIC ${CMAKE_C_FLAGS}")
+    set (CMAKE_CXX_FLAGS "-fPIC ${CMAKE_CXX_FLAGS}")
+  endif ()
+endif ()
 
-INSTALL ( TARGETS nlopt
-          RUNTIME DESTINATION bin
-          LIBRARY DESTINATION lib${LIB_SUFFIX}
-          ARCHIVE DESTINATION lib${LIB_SUFFIX}
+install ( TARGETS ${nlopt_lib}
+          EXPORT  NLoptLibraryDepends
+          RUNTIME DESTINATION ${RELATIVE_INSTALL_BIN_DIR}
+          LIBRARY DESTINATION ${RELATIVE_INSTALL_LIB_DIR}
+          ARCHIVE DESTINATION ${RELATIVE_INSTALL_LIB_DIR}
         )
 
+add_subdirectory (api)
+
+if (BUILD_PYTHON)
+  find_package (PythonInterp)
+  find_package (PythonLibs)
+  find_package (NumPy)
+endif ()
+
+if (PYTHONINTERP_FOUND)
+  execute_process ( COMMAND ${PYTHON_EXECUTABLE} -c "from distutils import sysconfig; print(sysconfig.get_python_lib(plat_specific=True, prefix='${CMAKE_INSTALL_PREFIX}'))"
+                    OUTPUT_VARIABLE _ABS_PYTHON_MODULE_PATH
+                    OUTPUT_STRIP_TRAILING_WHITESPACE )
+
+  get_filename_component (_ABS_PYTHON_MODULE_PATH ${_ABS_PYTHON_MODULE_PATH} ABSOLUTE)
+  file (RELATIVE_PATH _REL_PYTHON_MODULE_PATH ${CMAKE_INSTALL_PREFIX} ${_ABS_PYTHON_MODULE_PATH})
+
+  set (PYTHON_MODULE_PATH ${_REL_PYTHON_MODULE_PATH})
+
+endif ()
+
+if (BUILD_GUILE)
+  find_package (Guile)
+endif ()
+
+if (USE_SWIG)
+  find_package (SWIG)
+endif ()
+
 add_subdirectory (swig)
+
+if (BUILD_OCTAVE)
+  find_package (Octave)
+endif ()
+
+if (OCTAVE_FOUND)
+  add_subdirectory (octave)
+endif ()
+
+enable_testing ()
+add_subdirectory (test)
+
+set (CPACK_PACKAGE_NAME          "${CMAKE_PROJECT_NAME}")
+set (CPACK_PACKAGE_VERSION_MAJOR "${NLOPT_MAJOR_VERSION}")
+set (CPACK_PACKAGE_VERSION_MINOR "${NLOPT_MINOR_VERSION}")
+set (CPACK_PACKAGE_VERSION_PATCH "${NLOPT_BUGFIX_VERSION}")
+set (CPACK_SOURCE_GENERATOR      "TBZ2;TGZ"    )
+set (CPACK_BINARY_STGZ           "OFF"        )
+set (CPACK_BINARY_TBZ2           "ON"         )
+set (CPACK_BINARY_TGZ            "ON"         )
+set (CPACK_BINARY_TZ             "OFF"        )
+set (CPACK_SOURCE_IGNORE_FILES ".git;/build;.*~;${CPACK_SOURCE_IGNORE_FILES}")
+
+set (CPACK_SOURCE_PACKAGE_FILE_NAME ${CPACK_PACKAGE_NAME}-${NLOPT_VERSION_STRING})
+
+include (CPack)
+
+# write .def file
+if (MINGW AND BUILD_SHARED_LIBS)

```

Comment:
This seems like it may not be necessary with cmake?  The reason I was building the `.def` file is that the previous build script created the `.dll` when cross-compiling but didn't create the corresponding `.lib` import library, and people needed the `.def` file to create the `.lib` as described http://ab-initio.mit.edu/wiki/index.php/NLopt_on_Windows

I'm hoping that cmake is able to build the import library itself?